openmdao/core/tests/test_problem.py

```diff
@@ -535,5 +535,108 @@ def test_relevance(self):
+                self.add_input('a', val=1.)
+                self.add_output('x', val=0.)
+
+            def apply_nonlinear(self, inputs, outputs, residuals):
+                residuals['x'] = np.exp(outputs['x']) - \
+                    inputs['a']**2 * outputs['x']**2
+
+            def linearize(self, inputs, outputs, jacobian):
+                jacobian['x', 'x'] = np.exp(outputs['x']) - \
+                    2 * inputs['a']**2 * outputs['x']
+                jacobian['x', 'a'] = -2 * inputs['a'] * outputs['x']**2
+
+
+        class Sub(Group):
+
+            def setup(self):
+                self.add_subsystem('comp', ImplSimple())
+
+                # This will not solve it
+                self.nonlinear_solver = NonlinearBlockGS()
+
+            def configure(self):
+                # This will not solve it either.
+                self.nonlinear_solver = NonlinearBlockGS()
+
+
+        class Super(Group):
+
+            def setup(self):
+                self.add_subsystem('sub', Sub())
+
+            def configure(self):
+                # This will solve it.
+                self.sub.nonlinear_solver = NewtonSolver()
+                self.sub.linear_solver = ScipyIterativeSolver()
+
+
+        top = Problem()
+        top.model = Super()
+
+        top.setup(check=False)
+
+        x = -0.5
+        a = np.abs(np.exp(0.5 * x) / x)
+
+        top['sub.comp.a'] = a
+        top.run_model()
+        assert_rel_error(self, top['sub.comp.x'], x)
+
+    def test_feature_system_configure(self):
+
+        class ImplSimple(ImplicitComponent):
+
+            def setup(self):
+                self.add_input('a', val=1.)
+                self.add_output('x', val=0.)
+
+            def apply_nonlinear(self, inputs, outputs, residuals):
+                residuals['x'] = np.exp(outputs['x']) - \
+                    inputs['a']**2 * outputs['x']**2
+
+            def linearize(self, inputs, outputs, jacobian):
+                jacobian['x', 'x'] = np.exp(outputs['x']) - \
+                    2 * inputs['a']**2 * outputs['x']
+                jacobian['x', 'a'] = -2 * inputs['a'] * outputs['x']**2
+
+
+        class Sub(Group):
+            def setup(self):
+                self.add_subsystem('comp', ImplSimple())
+
+            def configure(self):
+                # This solver won't solve the sytem. We want
+                # to override it in the parent.
+                self.nonlinear_solver = NonlinearBlockGS()
+
+
+        class Super(Group):
+            def setup(self):
+                self.add_subsystem('sub', Sub())
+
+            def configure(self):
+                # This will solve it.
+                self.sub.nonlinear_solver = NewtonSolver()
+                self.sub.linear_solver = ScipyIterativeSolver()
+
+
+        top = Problem()
+        top.model = Super()
+
+        top.setup(check=False)
+
+        top['sub.comp.a'] = 1.5576015661428098
+        top.run_model()
+        assert_rel_error(self, top['sub.comp.x'], -0.5)
```

Comment:
also, the test could be a bit faster if we didn't actually have to run the solver. Since we're not testing solve behavior here, its not necessary and would even, I think make a better more unit-like test if we didn't run the solver